src/Calculator/Views/GraphingCalculator/EquationInputArea.xaml.cpp

```diff
@@ -128,26 +128,34 @@ void EquationInputArea::InputTextBox_Submitted(Object ^ sender, EquationSubmissi
 
 void EquationInputArea::FocusEquationTextBox(EquationViewModel ^ equation)
 {
-    auto nextContainer = EquationInputList->ContainerFromItem(equation);
-    if (nextContainer == nullptr)
-    {
-        return;
-    }
-    auto listviewItem = dynamic_cast<ListViewItem ^>(nextContainer);
-    if (listviewItem == nullptr)
+    unsigned int index;
+    if (!Equations->IndexOf(equation, &index) || index < 0)
     {
         return;
     }
-    auto equationInput = VisualTree::FindDescendantByName(nextContainer, "EquationInputButton");
-    if (equationInput == nullptr)
+    auto container = EquationInputList->TryGetElement(index);
+    if (container == nullptr)
     {
         return;
     }
-    auto equationTextBox = dynamic_cast<EquationTextBox ^>(equationInput);
+    auto equationTextBox = dynamic_cast<EquationTextBox ^>(container);
     if (equationTextBox != nullptr)
     {
         equationTextBox->FocusTextBox();
     }
+    else
+    {
+        auto equationInput = VisualTree::FindDescendantByName(container, "EquationInputButton");
```

Comment:
When there are variables and I type a new equation into the ghost text box, focus no longer goes to the new ghost textbox

Reply:
I can't repro this one, perhaps fixed by another fix I made. Let me know if you still have this issue